### PR TITLE
Function Spacing to include "Expanded"

### PR DESCRIPTION
**Pull request by euans**
This is a slight update to add an "extra-spaced" function formatting that separates the function name from the arguments (making it a bit easier to read).

### CODE REVIEW

**Comment by jcberquist** [comment]
Hi, thanks for the pull request. As it stands right now, you are using the `function_declaration.group_to_block_spacing` and `function_anonymous.group_to_block_spacing` settings for this. But those settings are for the spacing from the function params `()` to the function body `{}`, not for the spacing from the `function` keyword or the function name to the params. There is an analogous setting for this with keywords, `keywords.spacing_to_group` (which is a boolean), and I think that setting would be needed to be added to the function settings in order to add this.

**Comment by euans** [comment]
John-
I've removed the "expanded" flags from the "group_to_block_spacing' in lieu of the "spacing_to_group" as added in function_anonymous.spacing_to_group and function_declaration.spacing_to_group.  I've tested it and it appears to have the same outcome as the previous solution.  I look forward to your feedback.

File: data/reference.json [hunk 1]
```diff
@@ -103,6 +103,13 @@
         },
         "type": "boolean"
     },
+    "function_anonymous.spacing_to_group": {
+        "description": "Whether to space a function name from following group.",
+        "example": {
+            "code": "if (true){}"
```
Comment (jcberquist) [inline data/reference.json:109]: Could you update the example to match the setting?

File: data/reference.json [hunk 2]
```diff
@@ -225,6 +232,13 @@
         },
         "type": "boolean"
     },
+    "function_declaration.spacing_to_group": {
+        "description": "Whether to space a function name from following group.",
+        "example": {
+            "code": "if (true){}"
```
Comment (jcberquist) [inline data/reference.json:238]: Also here.

File: models/cfscript/Functions.cfc [hunk 3]
```diff
@@ -27,7 +27,12 @@ component {
             }
 
             var formattedText = words.toList(' ');
-            var rootSettingKey = token[2].last() == 'entity.name.function.cfml' ? 'function_declaration' : 'function_anonymous';
+            var rootSettingKey = token[2].last() == 'entity.name.function.cfml' ? 'function_declaration' : '    function_anonymous';
```
Comment (jcberquist) [inline models/cfscript/Functions.cfc:30]: It looks like there is extra white space in the setting key here?

File: models/cfscript/Functions.cfc [hunk 4]
```diff
@@ -69,6 +74,11 @@ component {
                 !attributesTxt.find(chr(10))
             ) {
                 formattedText = formattedText & ' ';
+            } else if (
+                settings['#rootSettingKey#.group_to_block_spacing'] == 'expanded' &&
+                !attributesTxt.find(chr(10))
+            ) {
+                formattedText = formattedText.replace('(', ' (') & ' ';
```
Comment (jcberquist) [inline models/cfscript/Functions.cfc:81]: Shouldn't this else block be removed now?

**Review by jcberquist — CHANGES_REQUESTED** [review]
I have added some comments to the code. Note that you don't need to update the `examples.json` and `reference.md` files directly. They are built from the `reference.json` file by the `build.cfc` task runner in the repo root. So you just need to update the `reference.json` file, and then run `task run build.cfc` with CommandBox.

File: data/reference.json (hunk 1 above)
Comment (euans) [reply data/reference.json]: Updated the example to represent the new functionality.

File: models/cfscript/Functions.cfc (hunk 3 above)
Comment (euans) [reply models/cfscript/Functions.cfc]: Removed excess spacing.

File: models/cfscript/Functions.cfc (hunk 4 above)
Comment (euans) [reply models/cfscript/Functions.cfc]: Removed old "expanced" if block as it is no longer utilized.

File: data/reference.json (hunk 2 above)
Comment (euans) [reply data/reference.json]: Updated documentation to reflect the new functionality.

**Review by jcberquist — APPROVED** [review]
Thanks, this looks good. I appreciate all the work!

**Pull request merged**